sdk-extensions/autoconfigure/src/testFullConfig/java/io/opentelemetry/sdk/autoconfigure/EndpointConfigurationTest.java

```diff
@@ -0,0 +1,110 @@
+
+@SuppressWarnings("InterruptedExceptionSwallowed")
+class EndpointConfigurationTest {
+
+  private static final BlockingQueue<ExportTraceServiceRequest> otlpTraceRequests =
+      new LinkedBlockingDeque<>();
+  private static final BlockingQueue<ExportMetricsServiceRequest> otlpMetricsRequests =
+      new LinkedBlockingDeque<>();
+
+  @RegisterExtension
+  public static final ServerExtension server =
+      new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+          sb.service(
+              GrpcService.builder()
+                  // OTLP spans
+                  .addService(
+                      new TraceServiceGrpc.TraceServiceImplBase() {
+                        @Override
+                        public void export(
+                            ExportTraceServiceRequest request,
+                            StreamObserver<ExportTraceServiceResponse> responseObserver) {
+                          otlpTraceRequests.add(request);
+                          responseObserver.onNext(ExportTraceServiceResponse.getDefaultInstance());
+                          responseObserver.onCompleted();
+                        }
+                      })
+                  // OTLP metrics
+                  .addService(
+                      new MetricsServiceGrpc.MetricsServiceImplBase() {
+                        @Override
+                        public void export(
+                            ExportMetricsServiceRequest request,
+                            StreamObserver<ExportMetricsServiceResponse> responseObserver) {
+                          if (request.getResourceMetricsCount() > 0) {
+                            otlpMetricsRequests.add(request);
+                          }
+                          responseObserver.onNext(
+                              ExportMetricsServiceResponse.getDefaultInstance());
+                          responseObserver.onCompleted();
+                        }
+                      })
+                  .useBlockingTaskExecutor(true)
+                  .build());
+        }
+      };
+
+  @BeforeEach
+  void setUp() {
+    otlpTraceRequests.clear();
+    otlpMetricsRequests.clear();
+    GlobalOpenTelemetry.resetForTest();
+  }
+
+  @Test
+  void configures() {
+    // Point "otel.exporter.otlp.endpoint" to wrong endpoint
+    System.setProperty("otel.exporter.otlp.endpoint", "http://localhost/wrong");
+
+    // Point "otel.exporter.otlp.traces.endpoint" to correct endpoint
+    System.setProperty(
+        "otel.exporter.otlp.traces.endpoint", "http://localhost:" + server.httpPort());
+
+    // Point "otel.exporter.otlp.metrics.endpoint" to correct endpoint
+    System.setProperty(
+        "otel.exporter.otlp.metrics.endpoint", "http://localhost:" + server.httpPort());
+
+    System.setProperty("otel.exporter.otlp.timeout", "10000");
+
+    GlobalOpenTelemetry.get().getTracer("test").spanBuilder("test").startSpan().end();
```

Comment:
Do we need to use the global instance for these tests?

Reply:
I guess one of the goal of `autoconfigure` module is to set global instance based on system properties/env variables.
So in the tests we set system properties and verify that global instance is configured properly.

Reply:
We should also confirm that the instance returned by the autoconfigure module is also configured properly, then. That should be the primary thing that we verify, and we can additionally verify the global as a secondary assertion, IMO.

Reply:
added test for non global configuration